data/modules/scripts/gamestore/init.lua

```diff
@@ -560,18 +560,41 @@ function Player.canBuyOffer(self, offer)
 				disabled = 1
 				disabledReason = "You already have an active XP boost."
 			end
+		elseif offer.type == GameStore.OfferTypes.OFFER_TYPE_HIRELING then
+			if self:getHirelingsCount() == 10 then
+				disabled = 1
+				disabledReason = "You have maximum of 10 hirelings."
```

Comment:
```suggestion
				disabledReason = "You already have bought the maximum number of allowed hirelings."
```
This way, if in the future it's implemented a configuration so the server owner can choose how much hirelings the players can have it will still be correct.